### PR TITLE
Html and css practice

### PR DESCRIPTION
**Pull request by cuteman18**
### HTML_and_CSS_practice

[Demo](https://cuteman18.github.io/#) |
[Code base](https://github.com/cuteman18/kottans-frontend/tree/main/HTML_and_CSS_practice)

The code is submitted in a dedicated feature branch.
Please, review.

> Will you have a replacement PR for Popup?

new PR #414

### CODE REVIEW

**Comment by lenadgit** [comment]
Could you please add a link to Popup instead of this link https://cuteman18.github.io/kottans-frontend/   for DEMO ?

**Comment by cuteman18** [comment]
Please find DEMO link updated.

File: submissions/cuteman18/HTML_and_CSS_practice/app.css [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+}
+.ref:hover, .bell:hover, .face:hover, .holder:hover {
+  opacity: 0.4;
+  cursor: pointer;
+}
+.dropDown, .button {
+  position: absolute;
+  overflow-y: auto;
+  clip: rect(0 0 0 0);
+}
```
Comment (lenadgit) [inline submissions/cuteman18/HTML_and_CSS_practice/app.css:80]: if you see this symbol at the end of the file on Github - it mean that the empty line is necessary 
empty line
Could you please add the empty line at the end of the file

File: submissions/cuteman18/HTML_and_CSS_practice/index.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
```
Comment (lenadgit) [inline submissions/cuteman18/HTML_and_CSS_practice/index.html:4]: Metatag Viewport is mandatory for mobile markup.
https://serpstat.com/ru/blog/chto-takoe-metateg-viewport/
Could you please add it?
`<meta name="viewport" content="width=device-width, initial-scale=1.0">`

File: submissions/cuteman18/HTML_and_CSS_practice/index.html [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/maps.png" alt="maps"/></span><span class="iconText">maps</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/files.png" alt="files"/></span><span class="iconText">files</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/contacts.png" alt="contacts"/></span><span class="iconText">contacts</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/photos.png" alt="photos"/></span><span class="iconText">photo</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/videos.png" alt="videos"/></span><span class="iconText">video</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/search.png" alt="search"/></span><span class="iconText">search</span></a></li>
+              <p>_________________________</p>
```
Comment (lenadgit) [inline submissions/cuteman18/HTML_and_CSS_practice/index.html:28]: you can use a tag `<hr>` as the divider  instead

**Review by lenadgit — CHANGES_REQUESTED** [review]

File: submissions/cuteman18/HTML_and_CSS_practice/app.css (hunk 1 above)
Comment (cuteman18) [reply submissions/cuteman18/HTML_and_CSS_practice/app.css:80]: added

File: submissions/cuteman18/HTML_and_CSS_practice/index.html (hunk 2 above)
Comment (cuteman18) [reply submissions/cuteman18/HTML_and_CSS_practice/index.html:4]: Thanks, added

File: submissions/cuteman18/HTML_and_CSS_practice/index.html (hunk 3 above)
Comment (cuteman18) [reply submissions/cuteman18/HTML_and_CSS_practice/index.html:28]: clear thanks, changed.

**Comment by cuteman18** [comment]
Changed, please review.

**Review by lenadgit — CHANGES_REQUESTED** [review]
Thanks for the fixes but could you please add these fixes to your branch "cuteman18:HTML_and_CSS_practice"  ?  
We will merge this branch.
view-source:https://cuteman18.github.io/#

Feel free taking the next course module while you submission is being reviewed

**Comment by cuteman18** [comment]
added index.html & app.css  files into my branch "cuteman18:HTML_and_CSS_practice" - please merge.

File: app.css [hunk 4]
```diff
@@ -0,0 +1,80 @@
+nav { font: 20px Copperplate; }
```
Comment (A-Ostrovnyy) [inline app.css:1]: Try to write code more consistent: In this line, you write code in one line but most of your code wrote in another way
```suggestion
nav { 
    font: 20px Copperplate;
}
```

File: app.css [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+.ref {
+  text-decoration: none;
+  color: darkblue;
+}
+.holder {
+  display: inline-block;
+  background-image: url("images/popup-button.png");
+  background-size: contain;
+}
+.face > img, .bell > img, .holder {
```
Comment (A-Ostrovnyy) [inline app.css:30]: Avoid writing too specific CSS selectors.

File: app.css [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+}
+.ref:hover, .bell:hover, .face:hover, .holder:hover {
+  opacity: 0.4;
+  cursor: pointer;
+}
+.dropDown, .button {
+  position: absolute;
+  overflow-y: auto;
+  clip: rect(0 0 0 0);
+}
```
Comment (A-Ostrovnyy) [inline app.css:80]: Add an empty line at the end of each file. [Reason.](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)

File: index.html [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    <meta charset="utf-8">
+    <title>HTML & CSS practice</title>
+    <link rel="stylesheet" type="text/css" href="app.css">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  </head>
+  <body>
+    <nav>
+      <ul class="navList">
+        <li class="inscription"><a href="#" class="ref">About</a></li>
+        <li class="inscriptionDeviate"><a href="https://youtu.be/xdYYN-4ttDg" class="ref" target="_blank">HOBBY</a></li>
```
Comment (A-Ostrovnyy) [inline index.html:13]: Avoid writing uppercase text in HTML. Visual representing must be implemented in CSS.
For example: when you change content your visual style must be saved and not depend on the markup.

File: index.html [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/files.png" alt="files"/></span><span class="iconText">files</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/contacts.png" alt="contacts"/></span><span class="iconText">contacts</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/photos.png" alt="photos"/></span><span class="iconText">photo</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/videos.png" alt="videos"/></span><span class="iconText">video</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/search.png" alt="search"/></span><span class="iconText">search</span></a></li>
+              <hr>
+              <input type="checkbox" class="button" id="extra"/><label for="extra" class="more">M O R E</label>
```
Comment (A-Ostrovnyy) [inline index.html:30]: Add styles on `:focus` state.

File: index.html [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/maps.png" alt="maps"/></span><span class="iconText">maps</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/files.png" alt="files"/></span><span class="iconText">files</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/contacts.png" alt="contacts"/></span><span class="iconText">contacts</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/photos.png" alt="photos"/></span><span class="iconText">photo</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/videos.png" alt="videos"/></span><span class="iconText">video</span></a></li>
+              <li class="icons"><a href="#" class="ref"><span class="icon"><img class="imgStyle" src="images/search.png" alt="search"/></span><span class="iconText">search</span></a></li>
+              <hr>
```
Comment (A-Ostrovnyy) [inline index.html:29]: Avoid adding visual elements to the markup.

File: app.css (hunk 4 above)
Comment (A-Ostrovnyy) [inline app.css:1]: What file is it, why he is not in your submission folder?
Remove all files not for review.

File: index.html [hunk 10]
```diff
@@ -0,0 +1,45 @@
+<!DOCTYPE html>
```
Comment (A-Ostrovnyy) [inline index.html:1]: What file is it, why he is not in your submission folder?

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work

**Comment by OleksiyRudenko** [comment]
There is a couple of issues with this PR that has to be fixed.

1. There are files that are not under `submissions` directory and will be added to the root of the repo. We do not want this.
image

2. There are files that aren't required
image

Please pay attention to the instructions:
image

**Comment by OleksiyRudenko** [comment]
@cuteman18 do you need any help with completing this?

**Comment by cuteman18** [comment]
> @cuteman18 do you need any help with completing this?

I am stuck with conflict when uploading changes, trying to resolve them

**Comment by OleksiyRudenko** [comment]
@cuteman18 looks like you have deleted your fork
image

There were similar cases with other students. Feel free yelling for help in the Students' chat. I am pretty sure those who faced similar issue would come up with ideas.

**Comment by cuteman18** [comment]
> @cuteman18 looks like you have deleted your fork
> image
> 
> There were similar cases with other students. Feel free yelling for help in the Students' chat. I am pretty sure those who faced similar issue would come up with ideas.

@OleksiyRudenko    I deleted that fork after I failed to submit my changes.   I re-created the fork and pushed the task again, under #412  -   can you please review it ?

**Comment by OleksiyRudenko** [comment]
@cuteman18 it would be faster if you ping reviewers from this PR. Also it will be helpful for them to know to what extent you fixed the issues based on their feedback.

**Comment by OleksiyRudenko** [comment]
@cuteman18 I am a bit confused. #412 says it is JS DOM task. This task was Popup.
Will you have a replacement PR for Popup?

**Comment by cuteman18** [comment]
@OleksiyRudenko     Sorry, it's a popup - renamed to "Popup"

**Comment by cuteman18** [comment]
> it would be faster if you ping reviewers from this PR. Also it will be helpful for them to know to what extent you fixed the issues based on their feedback.

Done.

**Comment by OleksiyRudenko** [comment]
Closing as replaced by #412 and consequently by #414

**Pull request closed without merging**